R/waapwlspetpeese.R

```diff
@@ -0,0 +1,610 @@
+        lowerCI = models[[estimator]]["(Intercept)", "lCI"],
+        upperCI = models[[estimator]]["(Intercept)", "uCI"]
+      ))
+  }
+  estimates <- estimates[nrow(estimates):1, ]
+
+  # handle NaN in the estimates
+  if (any(c(is.nan(estimates[,"mean"]), is.nan(estimates[,"lowerCI"]), is.nan(estimates[,"upperCI"]))))
+    plotEstimates$setError(gettext("The figure could not be created since one of the estimates is not a number."))
+
+
+  # make the plot happen
+  plot <- ggplot2::ggplot()
+
+  plot <- plot + ggplot2::geom_errorbarh(
+    ggplot2::aes(
+      xmin = estimates[,"lowerCI"],
+      xmax = estimates[,"upperCI"],
+      y    = 1:nrow(estimates)
+    ),
+    height = 0.3)
+  plot   <- plot + ggplot2::geom_point(
+    ggplot2::aes(
+      x = estimates[,"mean"],
+      y = 1:nrow(estimates)),
+    shape = 15)
+  plot <- plot + ggplot2::geom_line(ggplot2::aes(x = c(0,0), y = c(.5, nrow(estimates) + 0.5)), linetype = "dotted")
+  plot <- plot + ggplot2::scale_x_continuous(
+    bquote("Mean Estimate"~.(if (options[["measures"]] == "correlation") bquote(rho) else bquote(mu))),
+    breaks = jaspGraphs::getPrettyAxisBreaks(range(c(0, estimates[,"lowerCI"], estimates[,"upperCI"]))),
+    limits = range(jaspGraphs::getPrettyAxisBreaks(range(c(0, estimates[,"lowerCI"], estimates[,"upperCI"])))))
+  plot <- plot + ggplot2::scale_y_continuous(
+    "",
+    breaks = 1:nrow(estimates),
+    labels = estimates[,"model"],
+    limits = c(0.5, nrow(estimates) + 0.5))
+  plot <- plot + ggplot2::theme(
+    axis.ticks.y = ggplot2::element_blank()
+  )
+
+  plot <- plot + jaspGraphs::geom_rangeframe(sides = "b") + jaspGraphs::themeJaspRaw()
+  plotEstimates$plotObject <- plot
+
+  return()
+}
+# effect size transformations
+.wwppTransformEstimates      <- function(fit, options) {
+
+  if (options[["measures"]] == "general" || length(fit) == 0)
+    return(fit)
+
+  # in the case that correlation input was used, this part will transform the results back
+  # from the estimation scale to the outcome scale
+  for (i in 1:nrow(fit)) {
+    fit["(Intercept)", "Std. Error"]                <- .maInvTransformSe(fit["(Intercept)", "Estimate"], fit["(Intercept)", "Std. Error"], options[["muTransform"]])
+    fit["(Intercept)", c("Estimate", "lCI", "uCI")] <- .maInvTransformEs(fit["(Intercept)", c("Estimate", "lCI", "uCI")],      options[["muTransform"]])
+  }
+
+  return(fit)
+}
+# some additional helpers
+.wwppAddCi                   <- function(summaryFit) {
+
+  summaryFit$lCI <- summaryFit[, "Estimate"] + qnorm(0.025) * summaryFit[, "Std. Error"]
+  summaryFit$uCI <- summaryFit[, "Estimate"] + qnorm(0.975) * summaryFit[, "Std. Error"]
+
+  return(summaryFit)
+}
+.wwppAddWaapFootnote         <- function(table, waap) {
+
+  if (is.null(waap))
+    return()
+  else if (length(waap) == 0)
+    table$addFootnote(symbol = gettext("Warning:") , gettextf("There were only %1$i adequatelly powered studies. WAAP was not estimated.", attr(waap, "nPowered")))
```

Comment:
Maybe put WAAP was not estimated before "there were only...". Not required.

Reply:
yes, that's indeed better. Thanks